TestGetAPISpecsRPCSuccess is flaky, sometimes fails to bind

Full log: https://api.travis-ci.org/jobs/245838707/log.txt?deansi=true

```
=== RUN   TestGetAPISpecsRPCSuccess
2017/06/22 15:34:10 gorpc.Server: [:9090]. Cannot listen to: [listen tcp :9090: bind: address already in use]
2017/06/22 15:34:10 gorpc.Server: [:9090]. Cannot accept new connection: [accept tcp [::]:9090: use of closed network connection]
2017/06/22 15:34:10 gorpc.Client: [:9090]. Cannot decode response: [read tcp 127.0.0.1:35700->127.0.0.1:9090: read: connection reset by peer]
[Jun 22 15:34:10]  WARN RPC Handler: GetApiDefinitions() returned nil, returning empty string
[Jun 22 15:34:10] ERROR Failed decode: unexpected end of JSON input
[Jun 22 15:34:10]  INFO main: Detected 0 APIs
--- FAIL: TestGetAPISpecsRPCSuccess (0.00s)
	api_definition_test.go:380: Should return array with one spec []
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x939f25]

goroutine 319 [running]:
testing.tRunner.func1(0xc4202a3790)
	/home/travis/.gimme/versions/go1.8.1.linux.amd64/src/testing/testing.go:622 +0x29d
panic(0xcd6380, 0x13b7710)
	/home/travis/.gimme/versions/go1.8.1.linux.amd64/src/runtime/panic.go:489 +0x2cf
github.com/TykTechnologies/tyk/vendor/github.com/lonelycode/gorpc.(*defaultListener).Close(0xc4207703f0, 0xc4208d4000, 0xe5b950)
	/home/travis/gopath/src/github.com/TykTechnologies/tyk/vendor/github.com/lonelycode/gorpc/transport.go:76 +0x25
github.com/TykTechnologies/tyk.stopRPCMock(0xc4203e1340)
	/home/travis/gopath/src/github.com/TykTechnologies/tyk/api_definition_test.go:336 +0xab
github.com/TykTechnologies/tyk.TestGetAPISpecsRPCSuccess(0xc4202a3790)
	/home/travis/gopath/src/github.com/TykTechnologies/tyk/api_definition_test.go:382 +0x12a
testing.tRunner(0xc4202a3790, 0xe5b530)
	/home/travis/.gimme/versions/go1.8.1.linux.amd64/src/testing/testing.go:657 +0x96
created by testing.(*T).Run
	/home/travis/.gimme/versions/go1.8.1.linux.amd64/src/testing/testing.go:697 +0x2ca
exit status 2
```

Not sure why this would happen. I assume we have all the ports to ourselves.

I can look into this once I'm done with the 2.3.7 stuff.

--- a/api_definition_test.go
+++ b/api_definition_test.go
@@ -321,7 +321,9 @@ func startRPCMock(dispatcher *gorpc.Dispatcher) *gorpc.Server {
 
 	globalConf.SlaveOptions.ConnectionString = server.Addr
 
-	go server.Serve()
+	if err := server.Start(); err != nil {
+		panic(err)
+	}
 
 	return server
 }

